axiom/tests/test_packages.c

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "nr_php_packages.h"
+#include "tlib_main.h"
+#include "util_memory.h"
+
+static void test_php_package_create_destroy(void) {
+  nr_php_package_t* package;
+  nr_php_package_t* null_package = NULL;
+
+  // Test: create new package and ensure it contains correct information
+  package = nr_php_package_create("Laravel", "8.83.27");
+
+  tlib_pass_if_not_null("create package", package);
+  tlib_pass_if_str_equal("test package name", "Laravel", package->package_name);
+  tlib_pass_if_str_equal("test package version", "8.83.27",
+                         package->package_version);
+
+  nr_php_package_destroy(package);
+  nr_php_package_destroy(null_package);
+
+  // Test: passing NULL pointer should not cause crash
+  nr_php_package_destroy(NULL);
+}
+
+static void test_php_adding_packages_to_hashmap(void) {
+  nr_php_package_t* package1;
+  nr_php_package_t* package2;
+  nr_php_package_t* package3;
+  nr_php_packages_t* hm = NULL;
+  int count;
+
+  // Test: create multiple new packages and add to hashmap
+  package1 = nr_php_package_create("Package One", "10.1.0");
+  package2 = nr_php_package_create("Package Two", "11.2.0");
+  package3 = nr_php_package_create("Package Three", "12.3.0");
+
```

Comment:
Add a test when first parameter is `NULL`:
```suggestion
  /* Should not crash: */
  nr_php_packages_add_package(NULL, package1);
```

Reply:
I have added this test and also added a test when the second parameter is NULL [f0c3107](https://github.com/newrelic/newrelic-php-agent/pull/754/commits/f0c31076077ee319dd972f252fdd0425127ab191)